contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -535,6 +678,20 @@ namespace eosiosystem {
                check( creator == suffix, "only suffix may create this account" );
             }
          }
+    
+         // check that the account does not match a blacklist pattern stored in `account_name_blacklist_table`
+         // -------------------------------------------------------------------------------------------------------
+         account_name_blacklist_table bl_table(get_self(), get_self().value);
+         auto itr = bl_table.begin();
+         bool present = (itr != bl_table.end());
+
+         if (present) {
+            const std::vector<name>& blacklist{itr->disallowed};
+            for (auto pattern : blacklist) {
+               check(name_allowed(new_account_name, pattern),
+                     "Account name " + new_account_name.to_string() + " creation disallowed by rule: " + pattern.to_string());
```

Comment:
Should the error message more explicit, saying the pattern cannot be a substring?